docs/DeveloperGuide.adoc

```diff
@@ -231,6 +231,48 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+=== Add Feature
+
+The add feature allows the user to add items to the `RestaurantBook`.
+It also populates the item added to the `RestaurantBook` with parameters supplied by the user.
+
+==== Current Implementation
+
+The add feature is implemented as four separate commands, one for adding each type of item: `addMember`, `addBooking`, `addIngredient`, `addStaff`.
+Each command is implemented as a `AddCommand` class. For example, the `addMember` command is implemented as the `AddMember` class.
+All `AddCommand` class inherit from the `Command` class.
+As the general flow of the various add commands are similar, the `addMember` command will be used as an example here.
```

Comment:
If going into detail, and using sequence diagram to illustrate, maybe can talk about Parser as well